pkg/reconciler/route/traffic/rollout.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// rollout.go contains the types and functions to deal with
+// gradual rollout of the new revision for a configuration target.
+// The types in this file are expected to be serialized as strings
+// and used as annotations for progressive rollout logic.
+
+package traffic
```

Comment:
just wondering out loud - everything in here is either prefixed or suffixed with `rollout` so I wonder if it's worth putting it in a `rollout` package which'd make all the names a little shorter. Might be overkill though and I guess we can always move it later 🤷.

Reply:
I did that at first, but it was about the same. I want to do a few More prs and see how much it is intertwined with traffic, then maybe move.